src/Sentry.Maui/SentryMauiAppBuilderExtensions.cs

```diff
@@ -99,13 +99,26 @@ private static void RegisterMauiEventsBinder(this MauiAppBuilder builder)
 
                     var platformApplication = application.Delegate as IPlatformApplication;
                     platformApplication?.HandleMauiEvents(bind: false);
+
+                    SentryMauiEventProcessor.InForeground = false;
                 });
+
+                lifecycle.OnActivated(application => SentryMauiEventProcessor.InForeground = true);
+
+                lifecycle.DidEnterBackground(application => SentryMauiEventProcessor.InForeground = false);
+                lifecycle.OnResignActivation(application => SentryMauiEventProcessor.InForeground = false);
             });
 #elif ANDROID
             events.AddAndroid(lifecycle =>
             {
                 lifecycle.OnApplicationCreating(application => (application as IPlatformApplication)?.HandleMauiEvents());
                 lifecycle.OnDestroy(application => (application as IPlatformApplication)?.HandleMauiEvents(bind: false));
+
+                lifecycle.OnResume(activity => SentryMauiEventProcessor.InForeground = true);
```

Comment:
Shouldn't MAUI deal with firing these evens on the right platforms or not? I expect this to work on iOS/MacCatalyst too

Reply:
MAUI fires lifecycle events for each platform. iOS callbacks are also handled here and this works on iOS too. Docs don't say anything about MacCatalyst: https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/app-lifecycle?view=net-maui-8.0#cross-platform-lifecycle-events